docs/designs/cli.md

```diff
@@ -0,0 +1,80 @@
+# ElasticDL Client CLI
+
+## Background
+
+ElasticDL is a Kubernetes-Native deep learning framework. As it runs
+distributed training/prediction/evaluation jobs in a cluster, we need a client
+to submit the job to the cluster. Currently we have one but it's tightly
+coupled with the main package. User need pip install the whole elasticdl
+package and lots of dependencies and it's too heavy.
+
+The main functionality of the client is *building image for ElasticDL job* and
+*submitting ElasticDL job*. It only depedent on docker and Kubernetes Api.
+In this doc, we are focusing on a light-weight client CLI of ElasticDL.
+
+## User Story
+
+1. User develops model in the IDE (VSCode/Emacs/PyCharm) and commit to a git
+repo (optional). The directory structure is as follows:
+
+```TEXT
+my_model_repo
+- layers
+- models
+- utils
+requirements.txt
+```
+
+2. Generate an Dockerfile.
```

Comment:
In my mind, `elasticdl zoo init [python|tensorflow|pytorch]` should 

1. create a Dockerfile with the specified base image, like the following

   ```dockerfile
   FROM python
   COPY . /zoo
   RUN python -m pip install -r -y /zoo/requirements.txt
   ```

1. and the `requirements.txt` file including ElasticDL
   ```text
   elasticdl
   ```

Reply:
In my opinion, user's requirement.txt contains the necessary dependency s/he can debug or test on the local machine (such as executing a model.call can pass). It doesn't need contain `elasticdl`.
I think we can generate one line `RUN pip install elasticdl` obviously in the Dockerfile.